### PR TITLE
horizon: Fix GitHub action problem with verify-range push in master

### PR DESCRIPTION
**Pull request by 2opremio**
See https://github.com/stellar/go/runs/5395864747?check_suite_focus=true

Followup to #4250

### CODE REVIEW

**Review by leighmcculloch — APPROVED** [review]

**Comment by 2opremio** [comment]
I've realized that the checkout was broken in PRs with a fork head. I've fixed that too

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -31,10 +34,10 @@ jobs:
         go: [1.17, 1.16.7, 1.18.0-rc1]
     runs-on: ${{ matrix.os }}
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
       with:
         # For pull requests, build and test the PR head not a merge of the PR with the destination.
-        ref: ${{ github.head_ref || github.ref }}
+        ref: ${{ ( github.event.pull_request.head.sha || github.ref }}
```
Comment (leighmcculloch) [inline .github/workflows/go.yml:40]:
```suggestion
        ref: ${{ github.event.pull_request.head.sha || github.ref }}
```

**Comment by 2opremio** [comment]
@leighmcculloch PTAL

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -14,7 +14,10 @@ jobs:
         go: [1.17, 1.18.0-rc1]
     runs-on: ${{ matrix.os }}
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
+      with:
+        # For pull requests, build and test the PR head not a merge of the PR with the destination.
+        ref: ${{ ( github.event.pull_request.head.sha || github.ref }}
```
Comment (leighmcculloch) [inline .github/workflows/go.yml:20]: Is this a typo? 
```suggestion
        ref: ${{ github.event.pull_request.head.sha || github.ref }}
```

Comment (2opremio) [reply .github/workflows/go.yml]: Yep, an oversight

**Comment by 2opremio** [comment]
@leighmcculloch PTAL

**Review by leighmcculloch — APPROVED** [review]

**Pull request merged**